satellite/marketingweb/server.go

```diff
@@ -102,32 +148,102 @@ func (s *Server) parseTemplates() (err error) {
 		filepath.Join(s.templateDir, "free-offers-modal.html"),
 	)
 
+	pageNotFoundFiles := append(s.commonPages(),
+		filepath.Join(s.templateDir, "page-not-found.html"),
+	)
+
+	internalErrorFiles := append(s.commonPages(),
+		filepath.Join(s.templateDir, "internal-server-error.html"),
+	)
+
+	badRequestFiles := append(s.commonPages(),
+		filepath.Join(s.templateDir, "err.html"),
+	)
+
 	s.templates.home, err = template.New("landingPage").ParseFiles(homeFiles...)
 	if err != nil {
 		return Error.Wrap(err)
 	}
 
-	pageNotFoundFiles := append(s.commonPages(),
-		filepath.Join(s.templateDir, "page-not-found.html"),
-	)
-
 	s.templates.pageNotFound, err = template.New("page-not-found").ParseFiles(pageNotFoundFiles...)
 	if err != nil {
 		return Error.Wrap(err)
 	}
 
-	internalErrorFiles := append(s.commonPages(),
-		filepath.Join(s.templateDir, "internal-server-error.html"),
-	)
-
 	s.templates.internalError, err = template.New("internal-server-error").ParseFiles(internalErrorFiles...)
 	if err != nil {
 		return Error.Wrap(err)
 	}
 
+	s.templates.badRequest, err = template.New("bad-request-error").ParseFiles(badRequestFiles...)
+	if err != nil {
+		return Error.Wrap(err)
+	}
+
 	return nil
 }
 
+// convertStringToTime formats form time input as time.Time.
+func convertStringToTime(value string) reflect.Value {
+	v, err := time.Parse("2006-01-02", value)
+	if err != nil {
+		// invalid decoder value
+		return reflect.Value{}
+	}
+	return reflect.ValueOf(v)
+}
+
+// parseOfferForm decodes POST form data into a new offer.
+func parseOfferForm(w http.ResponseWriter, req *http.Request) (o rewards.NewOffer, e error) {
+	err := req.ParseForm()
+	if err != nil {
+		return o, err
+	}
+
+	if err := decoder.Decode(&o, req.PostForm); err != nil {
+		return o, err
+	}
+	return o, nil
+}
+
+// CreateOffer handles requests to create new offers.
+func (s *Server) CreateOffer(w http.ResponseWriter, req *http.Request) {
+	offer, err := parseOfferForm(w, req)
+	if err != nil {
+		s.log.Error("failed to convert form to struct", zap.Error(err))
+		s.serveBadRequest(w, req, err)
+		return
+	}
+	inviteeCredits := offer.InviteeCreditInCents
+	awardCredits := offer.AwardCreditInCents
+
+	offer.Status = rewards.Active
+	offerType := mux.Vars(req)["offer_type"]
+
+	offer.AwardCreditInCents = rewards.ToCents(awardCredits)
```

Comment:
This converting needs to happen before `decoder`, otherwise it won't be able to parse it into our struct